backend/populate_catalog/task.py

```diff
@@ -18,11 +30,82 @@
 CATALOG_KEY = "catalog.json"
 CONTENTS_KEY = "Contents"
 RECORDS_KEY = "Records"
+MESSAGE_ATTRIBUTES_KEY = "messageAttributes"
+
+LOGGER = set_up_logging(__name__)
 
 
 def lambda_handler(event: JsonObject, _context: bytes) -> JsonObject:
     """Main Lambda entry point."""
 
+    LOGGER.debug(dumps({EVENT_KEY: event}))
+
+    for message in event[RECORDS_KEY]:
+        if (
+            message[MESSAGE_ATTRIBUTES_KEY][MESSAGE_ATTRIBUTE_TYPE_KEY][STRING_VALUE_KEY_LOWER]
+            == MESSAGE_ATTRIBUTE_TYPE_ROOT
+        ):
+            handle_root(message[BODY_KEY])
+        elif (
+            message[MESSAGE_ATTRIBUTES_KEY][MESSAGE_ATTRIBUTE_TYPE_KEY][STRING_VALUE_KEY_LOWER]
+            == MESSAGE_ATTRIBUTE_TYPE_DATASET
+        ):
+            handle_dataset(message[BODY_KEY])
+        else:
+            raise UnhandledSQSMessageException("Unhandled SQS message type")
+
+    return {}
+
+
+class UnhandledSQSMessageException(Exception):
+    pass
+
+
+class GeostoreSTACLayoutStrategy(HrefLayoutStrategy):
+    def get_catalog_href(self, cat: Catalog, parent_dir: str, is_root: bool) -> str:
+        original_path = cat.get_self_href().split("/")
+        if is_root:
+            cat_root = parent_dir
+        else:
+            cat_root = os.path.join(parent_dir, original_path[-2])
+
+        return os.path.join(cat_root, original_path[-1])
+
+    def get_collection_href(self, col: Collection, parent_dir: str, is_root: bool) -> str:
+        original_path = col.get_self_href().split("/")
+        if is_root:
+            col_root = parent_dir
+        else:
+            col_root = os.path.join(parent_dir, original_path[-2])
+
+        return os.path.join(col_root, original_path[-1])
+
+    def get_item_href(self, item: Item, parent_dir: str) -> str:
+        original_path = item.get_self_href().split("/")
+        return os.path.join(parent_dir, original_path[-1])
+
+
+def handle_dataset(version_metadata_key: str) -> None:
+    """Handle writing a new dataset version to the dataset catalog"""
+    storage_bucket_path = f"{S3_URL_PREFIX}{ResourceName.STORAGE_BUCKET_NAME.value}"
+    dataset_prefix = version_metadata_key.split("/", maxsplit=2)[0]
```

Comment:
`maxsplit=2` gives you up to *three* parts. Since we only care about the first one I think we can use `maxsplit=1` here.